pkgs/build-support/oci/default.nix

```diff
@@ -0,0 +1,81 @@
+{
+  lib,
+  stdenvNoCC,
+  skopeo,
+  dockerTools,
+  nixosTests,
+}:
+
+rec {
+  buildContainer = throw "`ociTools.buildContainer` is removed. Use `ociTools.buildImage` instead.";
+
+  /*
+    toOCIImage is a generic conversion tool for converting Docker-style tarballs
+    into OCI images (directories or tarballs).
+
+    # Inputs
+    - `docker-tarball` (drv): The Docker-style tarball to convert. In most cases, this will
+      be the output of a `dockerTools.buildImage` call.
+    - `name` (string): The name of the resulting OCI image. If not provided, the name of the
+      Docker-style tarball will be used (with the `.tar.gz` suffix removed).
+    - `outputFormat` (string): The format of the resulting OCI image. Must be one of:
+      - `directory`: The output will be an OCI directory.
+      - `tarball`: The output will be a tarball of an OCI directory.
+  */
+  toOCIImage =
```

Comment:
I guess we can use the new stuff here https://github.com/NixOS/nixpkgs/pull/234651 to make this builder be overridable using finalAttrs pattern?

Reply:
Using `lib.mkExtendDerivation`? I'm not exactly sure which the base build helper would be in that case. But I can't say I understood that linked PR a 100% yet.

Reply:
Perhaps @ShamrockLee can help here?

Reply:
@msanft Let me know if this is helping.

Reply:
Thanks! This was helpful. Integrated it now.